### PR TITLE
Add AdjustableArrowCap tests and fix compat bugs

### PR DESCRIPTION
**Pull request by hughbe**
Builds on https://github.com/mono/libgdiplus/pull/99 to avoid merge conflicts

### CODE REVIEW

File: src/customlinecap.c [hunk 1]
```diff
@@ -49,13 +49,12 @@ void
...
 	cap->vtable = vt;
-	cap->base_cap = LineCapFlat;
+	cap->base_cap = LineCapTriangle;
 	cap->start_cap = LineCapFlat;
 	cap->end_cap = LineCapFlat;
 	cap->stroke_join = LineJoinMiter;
 	cap->base_inset = 0.0;
-	/* LAMESPEC: Default value is documented as 1.0, but actually it is 0.0 */
-	cap->width_scale = 0.0;
+	cap->width_scale = 1.0;
```
Comment (hughbe) [inline src/customlinecap.c]: This and above now match GDI+ behaviour

File: src/customlinecap.h [hunk 2]
```diff
@@ -14,6 +14,7 @@
 GpStatus WINGDIPAPI GdipCreateCustomLineCap (GpPath *fillPath, GpPath *strokePath, GpLineCap baseCap, float baseInset, GpCustomLineCap **customCap);
 GpStatus WINGDIPAPI GdipDeleteCustomLineCap (GpCustomLineCap *customCap);
 GpStatus WINGDIPAPI GdipCloneCustomLineCap (GpCustomLineCap *customCap, GpCustomLineCap **clonedCap);
+GpStatus WINGDIPAPI GdipGetCustomLineCapType( GpCustomLineCap* customCap, CustomLineCapType* capType);
```
Comment (akoeplinger) [inline src/customlinecap.h]: nit: space before the `(`, not after :)

Comment (akoeplinger) [reply src/customlinecap.h]: also `*` should be stuck to the argument, not the type, like everywhere else: `GpCustomLineCap *customCap`

File: src/customlinecap.c [hunk 3]
```diff
@@ -349,6 +348,16 @@ GdipCloneCustomLineCap (GpCustomLineCap *customCap, GpCustomLineCap **clonedCap)
 }
 
 GpStatus WINGDIPAPI
+GdipGetCustomLineCapType (GpCustomLineCap* customCap, CustomLineCapType* capType)
```
Comment (akoeplinger) [inline src/customlinecap.c]: see comment about `*` placement below.

File: src/adjustablearrowcap.c [hunk 4]
```diff
@@ -169,6 +169,11 @@ GdipCreateAdjustableArrowCap (REAL height, REAL width, BOOL isFilled, GpAdjustab
 	cap->width = width;
 	cap->height = height;
 
+	if (height == 0)
```
Comment (akoeplinger) [inline src/adjustablearrowcap.c]: shouldn't it check for width == 0 instead? otherwise we could have a divide by zero below.

Comment (hughbe) [reply src/adjustablearrowcap.c]: Thanks! This is totally a bug, my bad. I added some more edge-case tests that would have failed without this.
